fondant/cli.py

```diff
@@ -130,17 +117,123 @@ def explore(args):
         )
 
     if not shutil.which("docker"):
-        logging.error("Docker runtime not found. Please install Docker and try again.")
+        logging.error(
+            "Docker runtime not found. Please install Docker and try again.",
+        )
+
+    run_explorer_app(
+        data_directory=args.data_directory,
+        container=args.container,
+        tag=args.tag,
+        port=args.port,
+        credentials=args.credentials,
+    )
+
+
+def register_compile(parent_parser):
+    parser = parent_parser.add_parser(
+        "compile",
+        help="Compile a fondant pipeline.",
+    )
+    parser.add_argument(
+        "pipeline",
+        help="Path to the fondant pipeline: path.to.module:instance",
+        type=pipeline_from_string,
+    )
+    # add a mutually exclusive group for the mode
+    mode_group = parser.add_mutually_exclusive_group(required=True)
+    mode_group.add_argument("--local", action="store_true")
+    mode_group.add_argument("--kubeflow", action="store_true")
+
+    parser.add_argument(
+        "--output-path",
+        "-o",
+        help="Output directory",
+        default="docker-compose.yml",
+    )
+    parser.add_argument(
+        "--extra-volumes",
+        help="Extra volumes to mount in containers",
+        nargs="+",
+    )
+
+    parser.set_defaults(func=compile)
+
+
+def compile(args):
+    if args.local:
+        compiler = DockerCompiler()
+        compiler.compile(
+            pipeline=args.pipeline,
+            extra_volumes=args.extra_volumes,
+            output_path=args.output_path,
+        )
+    elif args.kubeflow:
+        msg = "Kubeflow compiler not implemented"
+        raise NotImplementedError(msg)
 
-    function_args = distill_arguments(args)
-    run_explorer_app(**function_args)
+
+def register_run(parent_parser):
+    parser = parent_parser.add_parser(
+        "run",
+        help="Run a fondant pipeline.",
+    )
+    parser.add_argument(
+        "ref",
+        help="""Reference to the pipeline to run, can be a path to a spec file or
+            a pipeline instance that will be compiled first""",
+        action="store",
+    )
+    # add a mutually exclusive group for the mode
+    mode_group = parser.add_mutually_exclusive_group(required=True)
+    mode_group.add_argument("--local", action="store_true")
+    mode_group.add_argument("--kubeflow", action="store_true")
+
+    parser.add_argument(
+        "--output-path",
+        "-o",
+        help="Output directory",
+        default="docker-compose.yml",
+    )
+    parser.add_argument(
+        "--extra-volumes",
+        help="Extra volumes to mount in containers",
+        nargs="+",
+    )
+    parser.set_defaults(func=run)
+
+
+def run(args):
+    if args.local:
+        try:
+            pipeline = pipeline_from_string(args.ref)
```

Comment:
We only need the try except around this line I think.

Reply:
Bump

Reply:
I could move the logic around but I believe this is the clearest. Try to see if it is an importstring if it is then compile it otherwise assume it is a reference to an already compiled pipeline.

Reply:
You can use 
```
try:
    ...
except:
    ...
else:
    ...
finally:
    ...
```